[Suryansh Kushwaha] iP

suryanshkushwaha opened the pull request:
# sillyBot
> _"The real risk is doing nothing."_ - Denis Waitley ([source](https://quotefancy.com/quote/793908/Denis-Waitley-The-real-risk-is-doing-nothing))

## Introduction
sillyBot is a fun and easy-to-use todo list manager designed to help you organize your tasks, deadlines, and events. With sillyBot, you can add, manage, and keep track of your tasks effortlessly. Whether you want to create a simple todo list or manage complex projects, sillyBot is here to assist you in a playful way. The best part: 
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

## Features
- [x] add tasks (todo, deadline, event)
- [x] delete tasks
- [x] mark/unmark tasks
- [x] deals with incomplete/invalid input exceptions
- [x] managing deadlines (coming soon)
- [x] reminders (coming soon)

## How to Use
1. download it from [here](https://github.com/suryanshkushwaha/ip/blob/master/duke.jar).
2. click "View raw" to download it.
3. open command window in the folder you downloaded.
4. run the command `java -jar Duke.jar`
5. let it manage your tasks for you 😉

And the BEST PART, it is **FREE**

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
new name

Commit:
echo

Commit:
add, list

Commit:
mark as Done

Commit:
ToDo, Event, Deadline

Commit:
Automated Text UI Testing

Commit:
Error Handling

Commit:
Delete

Commit:
refactor: created files/classes to follow OOPS

Commit:
fix: minor bug fixes (chatbot replies)

Commit:
added OOP

Commit:
added: Storage.java

Commit:
added: save/load feature

Commit:
added save/load feature

Commit:
added dates and times feature

Commit:
Merge branch 'branch-Level-8'
Added: Level 8

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle
Merge branch 'add-gradle-support' from origin

Commit:
Added: A Gradle

Commit:
Merge branch 'branch-A-Gradle'
Merge: Branch A Gradle

Commit:
added: JUnit

Commit:
Merge branch 'branch-A-JUnit'
Merge: Branch JUnit

Commit:
Minor tweaks to comply with Coding Standards

Commit:
updated java docs for all exceptions classes

Commit:
java docs for all task classes

Commit:
java docs for remaining classes

Commit:
minor tweaks to comply with coding standards

Commit:
added the find feature

Commit:
minor fix

Commit:
Merge branch 'branch-A-CodingStandard'
Added Coding Standards

Commit:
minor fix

Commit:
added Lvl 9

tyouwei commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,122 @@
+            break;
+        case "bye":
+            try {
+                storage.save(tasks);
```
Redundant line. It's not necessary to save. Tasks should be saved after every CRUD command is run.

suryanshkushwaha replied:
I thought to be on safer side by adding save option here as well. In case of any file handling error.

tyouwei commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+     */
+    public void showInvalidIndex() {
+        separatorLines();
+        System.out.println("You silly ah! That does not even exist! Wake up!!");
```
wah, fierce, I verry scared. -1 point

suryanshkushwaha replied:
Haha! I shall make it less fierce maybe.

tyouwei commented on src/main/java/duke/DatesAndTimesFormatter.java:
```diff
@@ -0,0 +1,35 @@
+public enum DatesAndTimesFormatter {
+    INPUT_FORMAT("yyyy-MM-dd"),
+    OUTPUT_FORMAT("MMM dd yyyy"),
+    SLASH_FORMAT("dd/MM/yyyy");
```
add more formats to make user input foolproof

suryanshkushwaha replied:
I agree. I shall add more formats to handle user input.

tyouwei reviewed (commented):
LGTM. Just some nits to fix.

Commit:
add duke.jar for GMFD

Commit:
add checkstyle with some minor fixes

Commit:
minor fix: changes made as per coding standards.

Commit:
Merge branch 'branch-A-CheckStyle'
add CheckStyle gradle checker

Commit:
add GUI as per lvl4

Commit:
Implement Level 10

Commit:
Merge branch 'branch-Level-10'
merge: Level 10 with master

Commit:
minor fix

AryanG01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+        try {
+            storage = new Storage(filePath);
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
```
Good Error catching here and creating a new empty list rather than terminating the program

AryanG01 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,146 @@
+    /**
+     * Prints the separator lines.
+     */
+    public void separatorLines() {
```
Good use of Abstraction

AryanG01 commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Represents a Todo object that is a Task.
+ */
+public class Todo extends Task {
```
A suggestion is if you would like to give Todo tasks date functionality as well

suryanshkushwaha replied:
But would it not make Todo more like the Deadline Task?

AryanG01 commented on src/test/java/duke/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,16 @@
+package duke.tasks;
```
Good job on doing testing for the different tasks

AryanG01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,132 @@
+        case "mark":
+            try {
+                Task task = tasks.get(Integer.parseInt(splittedCommands[1]) - 1);
+                if (Objects.equals(task.getStatusIcon(), "X")) {
```
Great to see you have added a functionality for this kind of actions, gives me ideas for my project as well

suryanshkushwaha replied:
I'm not sure which options are you referring to here, hehe

AryanG01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,132 @@
+            }
+            command = "bye";
+            break;
+        default:
```
Why do you perform checks here instead of just directly giving back an error warning message that such an action doesn't exist

suryanshkushwaha replied:
It is because the error can be either Input or Incomplete error so it would be better to check here

adhigop13 commented on src/main/java/duke/Parser.java:
Great use of JavaDocs! Makes it very clear for the reader to understand!

adhigop13 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,64 @@
+        this.filePath = filePath;
+        this.file = new File(filePath);
+        if (!file.exists()) {
+            System.out.println("File does not exist. Creating file...");
```
Perhaps the print statements from here can be moved to Ui class for more encapsulation?

suryanshkushwaha replied:
I agree. However, it was a bit hard to abstract them properly.

adhigop13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,146 @@
+        String output = "For some reason I believe you are gonna do that!\n" + task + "\n"
+                + "Now you have " + Task.getTaskCount() + " tasks in the list.";
+        System.out.println(output);
+        return output;
```
Perhaps these methods don't have to return and print the same thing?

suryanshkushwaha replied:
yea, I agree. I left the printing part for the CLI interface. I shall remove it.

adhigop13 commented on src/main/java/duke/exceptions/WrongMarkException.java:
Great use of Exceptions and creating multiple subclasses to handle different types of exceptions! 👍

suryanshkushwaha replied:
Thanks.

Commit:
add var args

Commit:
Merge branch 'branch-A-Varargs'
merge Branch-A-Varargs

Commit:
Modify duke.jar

Commit:
Add assert functionality

Commit:
Minor changes

Commit:
Refactor code

suryanshkushwaha committed:
Merge pull request #3 from suryanshkushwaha/branch-A-Assertions

Improve code robustness with assertions in branch-A-Assertions

- Added assertions to validate input conditions in several classes.
- Ensured tasks are not null when added to TaskList.
- Checked index validity before deleting tasks.

These changes enhance code reliability and error handling.

Commit:
Rectify merge conflicts

suryanshkushwaha committed:
Merge pull request #2 from suryanshkushwaha/branch-A-CodeQuality

Refactor code

- Introduced a new 'command' package to centralize command handling.
- Applied coding standards to all classes, ensuring consistent formatting.
- Organized and documented code for improved readability and maintainability.

These changes enhance code quality and adhere to coding standards.

Commit:
Delete duke folder

Commit:
Add gradle.yml file

Commit:
Merge branch 'branch-A-CI'

Commit:
Add Help command

Commit:
Merge branch 'branch-C-Help'

Commit:
Improve GUI

suryanshkushwaha committed:
Merge pull request #4 from suryanshkushwaha/branch-BetterGUI

Improve GUI

Commit:
Refine code documentation, enhance class naming, and resolve delete command bug

- Updated Java documentation for improved clarity and understanding.
- Renamed classes for better logical organization and readability.
- Fixed a bug related to the delete command, ensuring it functions as expected.

Commit:
Minor fix

Commit:
Add README.md file

- Include user guide providing an introduction, feature list, and usage information.
- Provide clear instructions on how to use the application.

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
Minor fix

Commit:
Minor fix
